.infrastructure/security/rbac.yml

```diff
@@ -0,0 +1,33 @@
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: secrets-reader
+  namespace: todoapp
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  namespace: todoapp # This is the namespace where the role will be created
+  name: secrets-lister-role # This is the name of the role
+rules:
+  - apiGroups: [""] # "" indicates the core API group, which includes all core APIs
+    resources: ["pods", "secrets"] # Indicates the resources that the role can access, in this case, pods
```

Comment:
The `resources` field includes `pods`, which might not be necessary if the intention is only to list and get `secrets`. Ensure that `pods` is intentionally included here.